internal/pkg/stream/cloudformation_test.go

```diff
@@ -240,6 +246,28 @@ func testStackStreamer_Fetch_WithError(t *testing.T) {
 	require.EqualError(t, err, "describe stack events phonetool-test: some error")
 }
 
+func testStackStreamer_Fetch_withThrottle(t *testing.T) {
+	// GIVEN
+	client := &mockCloudFormation{
+		err: awserr.New("RequestThrottled", "throttle err", errors.New("abc")),
+	}
+	streamer := &StackStreamer{
+		client:                *client,
+		clock:                 fakeClock{fakeNow: time.Now()},
```

Comment:
The cool thing about having a `fakeClock` now is we can pass some random date instead of `time.Now`! 
```suggestion
		clock:                 fakeClock{fakeNow: time.Date(2020, time.November, 23, 16, 0, 0, 0, time.UTC)},
```
and then below compare exactly `nextDate` received instead of using `Before` which is less precise:
```go
require.True(t, time.Date(2020, time.November, 23, 16, 8, 0, 0, time.UTC), nextDate)
```
This way we can validate that we indeed set the nextDate to be 8 seconds later 🎉

Reply:
We would need to update the structs to also have a random function that's mockable, so for example:

```go
type StackStreamer struct {
   clock clock
   rand  func(int) int
}

func NewStackStreamer(...) *StackStreamer {
   return &StackStreamer {
      rand: rand.Intn,
   }
} 
```
And then we can swap the `streamer.rand` field in the tests to be:
```go
&StackStreamer {
   rand: func(n int) { return n },
}
```
This way we'll get to validate that we do double it every time.